WallsAndHoles/tilemaptoolmanager.cpp

```diff
@@ -0,0 +1,25 @@
+#include "tilemaptoolmanager.h"
+
+TileMapToolManager::TileMapToolManager(QObject *parent)
+    : ToolManager(parent) {}
+
+void TileMapToolManager::setTileMap(TileMap *tileMap)
+{
+    for (AbstractToolP t : mTools)
+        static_cast<AbstractTileMapTool *>(t.data())->setTileMap(tileMap);
```

Comment:
I recommend putting these static casts away into a helper method. It will make the code much clearer.